doc/source/whatsnew/v1.0.1.rst

```diff
@@ -55,6 +55,9 @@ Bug fixes
 
 - Plotting tz-aware timeseries no longer gives UserWarning (:issue:`31205`)
 
+**Interval**
+
+- Bug in :meth:`arrays.IntervalArray.shift` raising a ``TypeError`` when shifting an interval array of integers or datetimes (:issue:`34195`)
```

Comment:
```suggestion
- Bug in :meth:`Series.shift` with ``interval`` dtype raising a ``TypeError`` when shifting an interval array of integers or datetimes (:issue:`34195`)
```

Reply:
(the docstring for IntervalArray.shift doesn't exist in sphinx + users will typically use Series.shift)